setup.py

```diff
@@ -69,7 +69,6 @@
     license='MIT',
     scripts=[],
     data_files=[],
-    setup_requires=["oldest-supported-numpy", "cython>=0.23"],
     install_requires=["numpy>=1.16", "scipy>=1.6"],
     python_requires=">=3.6",
```

Comment:
The `requires-python` metadata here is incorrect compared to reality as [`pot` `v0.9.3` only provides wheels for Python 3.7+](https://pypi.org/project/POT/0.9.3/#files) and the tests are only for Python 3.8+ 

https://github.com/PythonOT/POT/blob/cba9c7b7d27b59edf49979c746e480dbce787bc0/.github/workflows/build_tests.yml#L25

As @henryiii has covered elsewhere (can't remember which blog) the purpose of `requires-python` (`python_requires` in `setuptools`) is to provide guards to keep older CPython versions from installing releases that could contain unrunnable code and to avoid backtracking unhelpfully in dependency resolves.

This should get updated, but this is also a separate (adjacent) scope issue than NumPy 2.0, so I'll open up PR #630 for this.

Reply:
thanks we need to cleanup stuff